examples/client_example/client_example.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
+"""Python Client Example."""
+
+# Copyright 2012 - 2017, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+import argparse
+import logging
+import os
+import shutil
+import sys
+from pathlib import Path
+
+from tuf.exceptions import RepositoryError
+from tuf.ngclient import Updater
+
+# define directory constants
+HOME_DIR = Path.home()  # user home dir
+DOWNLOAD_DIR = "./downloads"  # download dir
+METADATA_DIR = f"{HOME_DIR}/.local/share/tuf_metadata_example"  # metadata dir
+CLIENT_EXAMPLE_DIR = os.path.dirname(os.path.abspath(__file__))  # example dir
+
+
+def init():
+    """
+    The function that initializes the TUF client infrastructure.
+
+    It creates the metadata directory and downloads the ``root.json``.
+    """
+
+    if not os.path.isdir(DOWNLOAD_DIR):
+        os.mkdir(DOWNLOAD_DIR)
+
+    print(f"Download directory [{DOWNLOAD_DIR}] is created")
+
+    if not os.path.isdir(METADATA_DIR):
+        os.makedirs(METADATA_DIR)
+
+    print(f"Metadata folder [{METADATA_DIR}] is created")
+
+    if not os.path.isfile(f"{METADATA_DIR}/root.json"):
+        shutil.copy(
+            f"{CLIENT_EXAMPLE_DIR}/1.root.json", f"{METADATA_DIR}/root.json"
+        )
+        print("Bootstrap initial root metadata")
+
+
+def tuf_updater():
+    """
+    The function that creates the TUF updater using ``tuf.ngclient``.
+    """
+    url = "http://127.0.0.1:8000"
+
+    updater = Updater(
+        repository_dir=METADATA_DIR,
+        metadata_base_url=f"{url}/metadata/",
+        target_base_url=f"{url}/targets/",
+        target_dir=DOWNLOAD_DIR,
+    )
+
+    return updater
+
+
+def download(target):
+    """
+    The function that downloads the target file using the TUF ``nglcient``
+    Updater.
+
+    The Updater refreshes the top-level metadata, get the target information,
+    verifies if the target is already cached, and in case it is not cached,
+    downloads the target file.
+    """
+
+    try:
+        updater = tuf_updater()
+
+    # if the metadata is not available, it will initialize the TUF client
+    except FileNotFoundError:
+        print("Client infrastructure not found")
+        print("Initializing the Client Infrastructure")
+        init()
```

Comment:
Ah so there is implicit init here as well -- I think it would look neater to always call init() but make init() cope with the fact that initialization may already be done. The `download()` does not need to worry or print anything about initialization, it's all handled in `init()`